blockies/src/main/java/pm/gnosis/blockies/BlockiesImageView.kt

```diff
@@ -0,0 +1,64 @@
+package pm.gnosis.blockies
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Path
+import android.util.AttributeSet
+import android.widget.ImageView
+
+class BlockiesImageView(context: Context, attributeSet: AttributeSet) : ImageView(context, attributeSet) {
+    private val canvasPaint = Paint().apply { style = Paint.Style.FILL }
+    private var dimen = 0.0f
+    private var offsetX = 0.0f
+    private var offsetY = 0.0f
+    private val path = Path()
+
+    private var blockies: Blockies? = null
+
+    fun setAddress(address: String) {
+        blockies = Blockies.fromAddress(address)
+        invalidate()
+    }
+
+    override fun onDraw(canvas: Canvas) {
+        super.onDraw(canvas)
+        blockies?.let { drawOnCanvas(canvas, it) }
+    }
+
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+        dimen = Math.min(measuredWidth, measuredHeight).toFloat()
+        offsetX = measuredWidth - dimen
+        offsetY = measuredHeight - dimen
+        path.addCircle(offsetX + (dimen / 2), offsetY + (dimen / 2), dimen / 2, Path.Direction.CCW)
```

Comment:
path.reset() needs to be called, else you are just adding circles over circles ;)